lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -1532,6 +1533,21 @@ void SwiftASTContext::ApplyWorkingDir(
   clang_argument.append(joined_path.begin(), joined_path.end());
 }
 
+void SwiftASTContext::ApplyDiagnosticOptions() {
+  const auto &opts = GetCompilerInvocation().getDiagnosticOptions();
+  if (opts.PrintDiagnosticNames) {
+    GetDiagnosticEngine().setPrintDiagnosticNames(true);
+  }
+  if (!opts.DiagnosticDocumentationPath.empty()) {
+    GetDiagnosticEngine().setDiagnosticDocumentationPath(
+      opts.DiagnosticDocumentationPath);
+  }
+  if (!opts.LocalizationCode.empty() && !opts.LocalizationPath.empty()) {
+    GetDiagnosticEngine().setLocalization(
+      opts.LocalizationCode, opts.LocalizationPath);
```

Comment:
Nit: if you haven't, could you run `clang-format` on your changes? I believe this line will be formatted differently.